smartcar/exception.py

```diff
@@ -0,0 +1,97 @@
+import json
+
+
+class SmartcarException(Exception):
+    """
+    All exceptions with this SDK will be a Smartcar Exception. v1.0 errors and
+    v2.0 errors are distinguished by the response body. "General" exceptions will
+    be raised as "SDK ERRORS".
+    """
+
+    def __init__(self, **kwargs):
+        # populate the error with passed in keys.
+        for key in kwargs.keys():
+            # format resolution: must be a dictionary with "type" and "url"
+            if key == "resolution":
+                if type((kwargs[key])) == str:
+                    formatted_resolution = {"type": kwargs[key], "url": None}
+                    self.__dict__[key] = formatted_resolution
+                if type(kwargs[key]) == dict:
```

Comment:
do you have tests for these cases?

Reply:
Not directly, nor are there unit tests for SmartcarException. 

However, in `test_exception.py`, there are a few lines that refer to the `resolution` attribute. That is because the tests run through all the weird cases of errors (OAuth, Vehicle State, Compatibility, Permission etc.), and this should cover the variances in returns for `resolution`.